android-components/components/lib/crash-sentry/src/main/java/mozilla/components/lib/crash/sentry/SentryService.kt

```diff
@@ -144,6 +148,20 @@ class SentryService(
         }
     }
 
+    private fun alterEventToBeUnhandled(event: SentryEvent) {
+        // Sentry now uses the `Mechanism` to determine whether or not an exception is
+        // handled. Any exception sent with `Sentry.captureException` is assumed to be handled
+        // by Sentry. We can attach a `UncaughtExceptionHandler` mechanism to the `SentryException`
+        // to correctly signal to Sentry that this is an uncaught exception.
+        // https://bugzilla.mozilla.org/show_bug.cgi?id=1835107
+        event.exceptions?.firstOrNull()?.let { sentryException ->
+            val mechanism = Mechanism()
+            mechanism.type = "UncaughtExceptionHandler"
+            mechanism.isHandled = false
+            sentryException.mechanism = mechanism
```

Comment:
💡 
```suggestion
            sentryException.mechanism = Mechanism().apply {
                type = "UncaughtExceptionHandler" // maybe this can be extracted as a constant
                isHandled = false
            }
```